src/vivarium/component.py

```diff
@@ -244,11 +248,7 @@ def initialization_requirements(self) -> Dict[str, List[str]]:
             list for a key implies no requirements for that key during
             initialization.
         """
-        return {
-            "requires_columns": [],
-            "requires_values": [],
-            "requires_streams": [],
-        }
+        return []
```

Comment:
It just occurrs to me that type-hinting dynamic things like this is confusing and I'm kinda surprised that mypy allows this to pass. Like, you're returning an empty list here but the type hints say it can also be a dict which simply isn't handled here.

Reply:
You're always allowed to say a function could return more than it actually will. A somewhat silly example, but imagine you have this function:
```python
def foo() -> bool:
    return True
```
This is obviously fine, even though it is not possible for `False` to be returned. A more precise type-hinting would be
```python
def foo() -> Literal[True]:
    return True
```